ipa-core/src/test_fixture/ipa.rs

```diff
@@ -41,13 +43,31 @@ impl GroupingKey for TestRawDataRecord {
     }
 }
 
+/// Insert `record` into `user_records`, maintaining timestamp order.
+///
+/// If there are existing records with the same timestamp, inserts the new record
+/// randomly in any position that maintains timestamp order.
+fn insert_sorted(user_records: &mut Vec<TestRawDataRecord>, record: TestRawDataRecord) {
```

Comment:
have you had a chance to test how long it takes to generate ~20M events with this change? I don't expect it to be terrible, but it would be great to know that for sure

Reply:
Running the revised event generation and `ipa_in_the_clear` with max_events_per_user up to 1,000 takes 5-10% longer than the previous version, which seems tolerable to me. With max_events_per_user of 10,000 the increase is more like 40%.